### ISSUE TITLE
Rapeseed detection using OpenEO data

### ISSUE THREAD

**Issue by shabarinath4** [WorldCereal/worldcereal-classification#176]
![Image](https://github.com/user-attachments/assets/ebdc19a7-65c8-4abb-8761-82390663e78f)
The data in this figure is from GEE. The orange cluster on the right is the rapeseed flowering event, which falls into a distinct cluster

![Image](https://github.com/user-attachments/assets/9e74ce3c-fd68-42cd-ba6d-da999189098a)
The data in this figure is from OpenEO. Here such a cluster is not observed. Needs to be further checked. In the GEE data I use 20 day intervals with max composite

Next step is to regenerate the data from OpenEO and repeat the experminet

**Comment by shabarinath4** [comment]
![image](https://github.com/user-attachments/assets/acffa850-01df-4283-a1e8-c0243b62453f)

`python scripts/extractions/extract.py POINT_NPA . ./notebooks/ESP_LVA_samples_2021_eo_data_sample.gpkg`

@kvantricht I am getting the following error today? Yesterday the jobs ran but I did'nt see any outputs in the output folder. Could there be any issue in my setup?

**Comment by kvantricht** [comment]
@shabarinath4 this can be an occasional outage on the catalogue side of CDSE, sadly it does happen. Can you simply retry and let me know if the issue persists?

**Comment by kvantricht** [comment]
As for the absence of outputs: was the job successfully finished? Does the workflow include the actual download of the result and does the output path correspond with your expectation? You can always go into the dashboard (https://openeo.dataspace.copernicus.eu/) and if the job finished you can also manually download the result there. But of course, we need to track down while the file is not on the expected location.

**Comment by shabarinath4** [comment]
ok got it. I will retry again in some time.

**Comment by shabarinath4** [comment]
@kvantricht is there a flag I need to set to download. It is the same workflow we setup the other day on the call . I was getting this before, 
![image](https://github.com/user-attachments/assets/5b866c43-dbcb-4e4f-9cc9-a3cf15370493)

**Comment by kvantricht** [comment]
Normally not I think. If you specified an output path, then it should automatically download the results to that path. From the screenshot it looks like that was already done. First time it should print the path where it stores the result actually. It's not saved to the `...\scripts\extractions` folder?

**Comment by shabarinath4** [comment]
[job_tracking.csv](https://github.com/user-attachments/files/18129601/job_tracking.csv)

@kvantricht  all jobs have the status failed thats why. I can check the description of the job id for the failure reason

**Comment by kvantricht** [comment]
@shabarinath4 I've posted this in our internal support channel. Checking your CSV, the status of all jobs in fact is `start_failed` which means processing didn't even start. I do seem to be able to find your jobs in our dashboard, although I couldn't find obvious errors myself. Stay tuned for an update from one of our colleagues (usually goes fast).

**Comment by shabarinath4** [comment]
@kvantricht  I think this was the issue. So the catalogue.py inside openeo_gfmap library uses a minx, miny, maxx, maxy. The job error was the min and max coordinates was the same so the API call is not able to create a valid bound. I the df['geometry'] cannot be POINT irrespective of point or patch extractions. 

`    minx, miny, maxx, maxy = bounds

    # The date format should be YYYY-MM-DD
    start_date = f"{temporal_extent.start_date}T00:00:00Z"
    end_date = f"{temporal_extent.end_date}T00:00:00Z"

    url = (
        f"https://catalogue.dataspace.copernicus.eu/resto/api/collections/"
        f"{collection}/search.json?box={minx},{miny},{maxx},{maxy}"
        f"&sortParam=startDate&maxRecords=1000&dataset=ESA-DATASET"
        f"&startDate={start_date}&completionDate={end_date}"
    )
    for key, value in additional_parameters.items():
        url += f"&{key}={value}"

    session = _request_session()
    response = session.get(url, timeout=60)`

**Comment by shabarinath4** [comment]
I added a small patch in the catalogue.py. Ideally need to buffer the POINT in the creation of the input df

`
    if minx == maxx:
        minx = minx - 0.0001
        maxx = maxx + 0.0001
    if miny == maxy:
        miny = miny - 0.0001
        maxy = maxy + 0.0001`

but after this the jobs were created

**Comment by shabarinath4** [comment]
but in the console few jobs finised, 6 out of 123. rest were stuck at 'created' status. So now I am running another script which is checking for 'created' jobs and starting it if the number of 'running' jobs is less than 2. The data is getting dowloaded. I will keep you posted once all done is dowloaded

**Comment by kvantricht** [comment]
> I added a small patch in the catalogue.py. Ideally need to buffer the POINT in the creation of the input df
> 
> ` if minx == maxx: minx = minx - 0.0001 maxx = maxx + 0.0001 if miny == maxy: miny = miny - 0.0001 maxy = maxy + 0.0001`
> 
> but after this the jobs were created

Which version of gfmap are you using? Normally this issue is already taken care of: https://github.com/Open-EO/openeo-gfmap/blob/main/src/openeo_gfmap/utils/catalogue.py#L207

**Comment by kvantricht** [comment]
> but in the console few jobs finised, 6 out of 123. rest were stuck at 'created' status. So now I am running another script which is checking for 'created' jobs and starting it if the number of 'running' jobs is less than 2. The data is getting dowloaded. I will keep you posted once all done is dowloaded

so normally you don't have to do all this manually if you're using the gfmap job manager. Are you implementing these things manually?

**Comment by shabarinath4** [comment]
> > I added a small patch in the catalogue.py. Ideally need to buffer the POINT in the creation of the input df
> > ` if minx == maxx: minx = minx - 0.0001 maxx = maxx + 0.0001 if miny == maxy: miny = miny - 0.0001 maxy = maxy + 0.0001`
> > but after this the jobs were created
> 
> Which version of gfmap are you using? Normally this issue is already taken care of: https://github.com/Open-EO/openeo-gfmap/blob/main/src/openeo_gfmap/utils/catalogue.py#L207

I cloned the sample-free approaches branch and the openeo_gfmap version is Version: 0.2.0

**Comment by shabarinath4** [comment]
> > but in the console few jobs finised, 6 out of 123. rest were stuck at 'created' status. So now I am running another script which is checking for 'created' jobs and starting it if the number of 'running' jobs is less than 2. The data is getting dowloaded. I will keep you posted once all done is dowloaded
> 
> so normally you don't have to do all this manually if you're using the gfmap job manager. Are you implementing these things manually?

yes all jobs got automatically launched, queued and several finished, but the rest were in created status. I am not sure if they were waiting for resources, but now a script is checking that and initiating it

**Comment by kvantricht** [comment]
then it's weird this buffering is not already there ..

**Comment by jdegerickx** [comment]
Update Shabri:
 
1. The optical based model is working as before now. 
2. Based on the cluster selection it can have very high precision, low recall or medium precision and recall. 
3. Depending on how we plan to use it this can be set. 
4. I have been thinking instead of using the sample free approaches directly for mapping, should we use it to extract high confidence samples to be integrated into training the main model. In this way we don't have to run inference for every sample free method, as these methods could be quite different from each other. Additionally there would not be the problem of majority voting etc among the data driven and sample free approach. 
5. The experiment would then be to test if adding high confidence samples derived from sample free approaches would improve the performance in a data sparse region. The challenge is however if the main model is already quite generasable and adding the samples doesn't improve performance, then how do we justify the value of sample free approaches?
6. One way I could think of is in the future years post release, these high confidence pixels could serve as a validation dataset to monitor any deterioration in model performance. Will be great to know what you think about it and then we can have a discussion on it.
